engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +60,53 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        if (!init.isDone()) {
+            return CompletableFuture.completedFuture(new ExecutionOutcome(Stage.ABORTED, null, "Container initialization was canceled before it started."));
+        }
+
+        return init.thenCompose(containerId -> {
+            if (result.isDone()) {
+                return CompletableFuture.completedFuture(result.join());
+            } else {
+                return CompletableFuture.supplyAsync(() -> {
+                    try {
+                        boolean isContainerRunning = isContainerRunning(containerId);
+                        if (isContainerRunning) {
+                            client.stopContainerCmd(containerId).exec();
+                            return new ExecutionOutcome(Stage.ABORTED, null, "Docker container aborted successfully:\n" + stdout);
+                        } else {
+                            return new ExecutionOutcome(Stage.ABORTED, null, "Docker container has already completed.");
+                        }
+                    } catch (NotFoundException e) {
+                        log.error("Container not found: {}", e.getMessage());
+                        return new ExecutionOutcome(Stage.ABORT, "Container not found: " + e.getMessage(), "Container not found: " + e.getMessage() + "\n" + stdout);
+                    } catch (DockerException e) {
+                        log.error("Error stopping the Docker container: {}", e.getMessage());
+                        return new ExecutionOutcome(Stage.ABORT, "Error aborting Docker container: " + e.getMessage(), "Error aborting Docker container: " + e.getMessage() + "\n" + stdout);
+                    }
+                }, executor);
+            }
+        });
+    }
+
+    private boolean isContainerRunning(String containerId) {
+        try {
+            InspectContainerResponse containerInfo = client.inspectContainerCmd(containerId).exec();
```

Comment:
There is no point in running a separate command to check state. You still have to deal with stop errors.